aspnetcore/mvc/controllers/routing.md

```diff
@@ -167,570 +235,656 @@ public class ProductsController : Controller
+[Areas](xref:mvc/controllers/areas) are an MVC feature used to organize related functionality into a group as a separate:
 
-The following example configures MVC to use the default conventional route and an *area route* for an area named `Blog`:
+* Routing namespace for controller actions.
+* Folder structure for views.
+
+Using areas allows an app to have multiple controllers with the same name, as long as they have different areas. Using areas creates a hierarchy for the purpose of routing by adding another route parameter, `area` to `controller` and `action`. This section discusses how routing interacts with areas. See [Areas](xref:mvc/controllers/areas) for details about how areas are used with views.
+
+The following example configures MVC to use the default conventional route and an `area` route for an `area` named `Blog`:
 
 [!code-csharp[](routing/samples/3.x/AreasRouting/Startup.cs?name=snippet1)]
-When matching a URL path like `/Manage/Users/AddUser`, the first route will produce the route values `{ area = Blog, controller = Users, action = AddUser }`. The `area` route value is produced by a default value for `area`, in fact the route created by `MapAreaControllerRoute` is equivalent to the following:
 
-[!code-csharp[](routing/samples/3.x/AreasRouting/Startup.cs?name=snippet2)]
+In the preceding code, <xref:Microsoft.AspNetCore.Builder.ControllerEndpointRouteBuilderExtensions.MapAreaControllerRoute*> is called to create the `"blog_route"`. The second parameter, `"Blog"`, is the area name.
+
+When matching a URL path like `/Manage/Users/AddUser`, the `"blog_route"` route generates the route values `{ area = Blog, controller = Users, action = AddUser }`. The `area` route value is produced by a default value for `area`. The route created by `MapAreaControllerRoute` is equivalent to the following:
+
+[!code-csharp[](routing/samples/3.x/AreasRouting/Startup2.cs?name=snippet2)]
 
 `MapAreaControllerRoute` creates a route using both a default value and constraint for `area` using the provided area name, in this case `Blog`. The default value ensures that the route always produces `{ area = Blog, ... }`, the constraint requires the value `{ area = Blog, ... }` for URL generation.
 
-> [!TIP]
-> Conventional routing is order-dependent. In general, routes with areas should be placed earlieras they're more specific than routes without an area.
+Conventional routing is order-dependent. In general, routes with areas should be placed earlier as they're more specific than routes without an area.
 
-Using the above example, the route values would match the following action:
+Usin
```

Comment:
Not really. It's an attempt to simplify a really complex concept. let's just take it out.